api/app/controllers/commits_controller.rb

```diff
@@ -18,6 +18,24 @@ def commits_over_time
     render(json: stats)
   end
 
+  def files_over_time
+    return render_repository_not_found unless current_repository
+
+    stats = RepositoryStatisticsService
+      .new(current_repository)
+      .file_modifications_by_date(start_date: start_date_filter, end_date: end_date_filter)
+      .map! do |filepath, total_additions, total_deletions, total_modifications|
+        {
+          filepath: filepath,
+          total_additions: total_additions,
+          total_deletions: total_deletions,
+          total_modifications: total_modifications
+        }
+      end
+
+    render(json: stats)
+  end
```

Comment:
We should move this to the `files_controller` since it's data related to files, not commits.

Reply:
done